packages/template-retail-react-app/app/pages/cart/index.jsx

```diff
@@ -307,11 +307,11 @@ const Cart = () => {
     }
 
     /********* Rendering  UI **********/
-    if (isLoading) {
+    if (isLoading || isProductsLoading) {
```

Comment:
If productIds is not defined, the isProductsLoading always true, so it will never show empty cart? 🤔

Reply:
sorry, why is `isProductsLoading` always true? i thought it is true only when the request is in-flight ?

Reply:
From the [docs](https://tanstack.com/query/v4/docs/react/reference/useQuery):
* `isLoading = status === 'loading'`
* `status` will be `'loading'` if there's no cached data and no query attempt was finished yet

If we need it, looks like there is a flag for `isInitialLoading`.

Reply:
@wjhsf `isInitialLoading` may good on first rendering, but we have the option to edit cart items, when it happens new productIds will be fetched, that means this loading state won't be useful for that

Reply:
reverted this change as it is not super relevant for this PR... I'm actually not sure why i have this change...